src/resmom/linux/alps.c

```diff
@@ -5817,15 +5817,61 @@ alps_cancel_reservation(job *pjob)
 {
 	char buf[1024];
 	basil_response_t *brp;
+	char	filename[MAXPATHLEN];
 
 	if (!pjob) {
 		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_NODE, LOG_NOTICE, __func__,
 			"Cannot cancel ALPS reservation, invalid job.");
 		return (-1);
 	}
+
+	/*
+	 * Get the name of the file that contains ALPS reservation ID
+	 * information.  We may need to read the file, or we just need to
+	 * unlink it.  Either way, we need the filename.
+	 */
+	snprintf(filename, MAXPATHLEN, "%s/aux/%s.alpsresv", pbs_conf.pbs_home_path, pjob->ji_qs.ji_jobid);
+	
+	if (pjob->ji_extended.ji_ext.ji_reservation < 0) {
+		/* 
+			* The job structure doesn't have ALPS reservation ID information.
+			* This can happen when we hit various race conditions.  To help
+			* with this issue, we have previously saved the ALPS resv ID
+			* in a file.
+			*
+			* The format of the file will consist of one line entry that
+			* has the string alps_resv_id a space and the ALPS reservation ID
+			* number.  We retrieve it here so the ALPS reservation can
+			* be canceled.
+			*/
+		FILE    *fp = NULL;
+		long    value;
+
+		fp = fopen(filename, "r");
+		if (fp == NULL) {
+			snprintf(log_buffer, sizeof(log_buffer), "Can't open the file %s", filename);
```

Comment:
I would prefer the log to explain the purpose of the file along with its name.

Reply:
Please help me understand how will explaining the purpose help user and please suggest how should I write the log message.

Reply:
how is the current log helping the user to understand anything? It seems like an FYI to the user and this information is not useful or helps the user to debug anything.
As an improvement, we can let the user know that there will be an orphan reservation in the system.